### PR TITLE
[Merged by Bors] - feat: port Data.Pi.Lex

### PR DESCRIPTION
**Pull request by javra**
mathlib3 SHA: d4f69d96f3532729da8ebb763f4bc26fcf640f06

### CODE REVIEW

File: Mathlib/Data/Pi/Lex.lean [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+instance {α : Type _} : ∀ [Inhabited α], Inhabited (Lex α) :=
+  @fun x => x
+
+/-- The lexicographic relation on `Π i : ι, β i`, where `ι` is ordered by `r`,
+  and each `β i` is ordered by `s`. -/
+protected def Lex (x y : ∀ i, β i) : Prop :=
+  ∃ i, (∀ j, r j i → x j = y j) ∧ s (x i) (y i)
+#align pi.lex Pi.Lex
+
+-- mathport name: «exprΠₗ , »
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Pi/Lex.lean:51]:
```suggestion
```

**Review by Ruben-VandeVelde — CHANGES_REQUESTED** [review]

File: Mathlib/Order/Basic.lean [hunk 2]
```diff
@@ -654,6 +654,23 @@ theorem update_le_update_iff {ι : Type u} {α : ι → Type v} [∀ i, Preorder
     Function.update x i a ≤ Function.update y i b ↔ a ≤ b ∧ ∀ (j) (_ : j ≠ i), x j ≤ y j := by
   simp (config := { contextual := true }) [update_le_iff]
 
+
+@[simp] lemma le_update_self_iff {ι : Type u} {α : ι → Type v} [∀ i, Preorder (α i)]
```
Comment (Ruben-VandeVelde) [inline Mathlib/Order/Basic.lean:658]: I've created  #1112 which conflicts with this. Let me know how you'd prefer to proceed.

Comment (javra) [reply Mathlib/Order/Basic.lean:658]: Go ahead with your one, I'll remove the `Mathlib.Order.Basic` stuff here, then.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**